### ISSUE TITLE
MySQL/MariaDB support for docker image OOTB

### ISSUE THREAD

**Issue by MenchenFive** [wger-project/wger#926]
## Use case

I am currently attempting to install wger in my homeserver via docker. Because I have other services running, like gitea or nextcloud, I wanted to reuse the existing database container I had by setting up a db and user for wger to use. After doing so, I set up the environment variable `DJANGO_DB_ENGINE` to `django.db.backends.mysql` per django [documentation](https://docs.djangoproject.com/en/4.0/ref/databases/#mariadb-notes), thinking it would work. But it didn't, since it seems that the mysql connector and its dependencies are missing from the wger image.

While it would be easy to spin up a postgre instance just for wger; some of the reasons of me not wanting to do that is performance (an extra database manager would be extra resources), and centralization: Docker makes it easy to spin up a db for each web service, but centralizing all services in just one dbms instance has its advantages too, like making maintenance easier since there is only one instance to dump for backups and take care of; though I do not want to turn this issue into a monolith vs microservices war.

I understand that you can't/shouldn't support every single DBMS out there, but mariadb is one of the major contestants along postgre (_and sqlite_) and allowing this choice would be more than welcome, I'm sure.

## Proposal

Include [mysqlclient](https://github.com/PyMySQL/mysqlclient) and its neccessary dependencies in the docker image.

**Comment by DorianHawkmoon** [comment]
I agree with it. I'm trying to install it without the docker, so manually installed the library dependency, and I had to import the library in __init.py__ but still doesn't seem to properly work.
Do you have plans for improving myqsl support?

**Comment by Faerbit** [comment]
Any news on this? I could whip up a PR if there is hope of it being merged.

**Comment by rolandgeider** [comment]
@Faerbit sure, do you  have something in mind for this? I'd prefer avoiding adding mysql as a new dependency that most will not use, but we could perhaps add this to the entrypoint or something similar

**Comment by Faerbit** [comment]
@rolandgeider I tried this on my own, and noticed that the migrations somehow fail with MariaDB anyway. So whatever the approach to include this would be, there isn't much value anyway, with the current state of things.

**Comment by rolandgeider** [comment]
Which one(s)? There is only one thing that is postgres specific, but that migration should only run on that db

**Comment by Faerbit** [comment]
@rolandgeider I'm not so sure entirely. This is the log:

[wger.log](https://github.com/user-attachments/files/22011802/wger.log)

(and I tried multiple times resetting the database and restarting the container. There could be of course some error with my setup, but I didn't find it, so I gave up)

**Comment by rolandgeider** [comment]
Interesting "Cannot delete or update a parent row: a foreign key constraint fails" on manager.0019_flexible_routines_migration, then some of the following ones also fail. But there's also "Table 'wger.django_site' doesn't exist" which is weird since that's something that comes directly from django, mhhh.

I wonder why any of this doesn't make problems with postgres?

**Comment by rolandgeider** [comment]
it's a bit late, but this was fixed and will be available with 2.5
